### ISSUE TITLE
Bug: when a MP4 file is appendTo() the DOM, callback of getBlob() isn't invoked after the file is downloaded

### ISSUE THREAD

**Issue by joshuajazleung** [webtorrent/webtorrent#1543]
The issue is similar to what I made https://github.com/webtorrent/webtorrent/issues/1542 a few days ago. But now the problem is related to getBlob() instead of getBlobURL()

```
     this.client.add(this.torrentId, torrent => {
                this.filename = path.basename(torrent.name, path.extname(torrent.name)) + '.zip';
                let zip = new JSZip();
                this.addedFiles = true;
                let hasVideoPlayed = false;

                torrent.files.forEach(file => {
                    this.files.push(file);

                    file.getBlobURL(function(err, url) {
                        if (err) {
                            console.log('error:' + err);
                        }

                        file.url = url;
                        console.log('url: ' + url);
                    });

                    if (file.name.endsWith('.mp4') && !hasVideoPlayed) {
                        file.appendTo(
                            '#videoList',
                            {
                                muted: true,
                                maxBlobLength: 2 * 1000 * 1000 * 1000 // 2 GB
                            },
                            function(err) {
                                if (err) {
                                    console.log(err);
                                }
                            }
                        );
                        hasVideoPlayed = true;
                    }

                    file.getBlob((err, blob) => {
                        downloadedFiles += 1;
                        console.log(downloadedFiles);
                        if (err) {
                            console.log(err);
                            return;
                        }

                        // add file to zip
                        zip.file(file.path, blob);

                        // start the download when all files have been added
                        if (downloadedFiles === torrent.files.length) {
                            if (torrent.files.length > 1) {
                                // generate the zip relative to the torrent folder
                                zip = zip.folder(torrent.name);
                            }
                            zip.generateAsync({ type: 'blob' }).then(blob => {
                                this.downloadURLs.direct = URL.createObjectURL(blob);
                                setTimeout(() => {
                                    URL.revokeObjectURL(this.downloadURLs.direct);
                                    this.downloadURLs.direct = '';
                                }, 30 * 1000);
                            }, console.error);
                        }
                    });
                });
            });
```

When the  `if (file.name.endsWith('.mp4') && !hasVideoPlayed) {}` block is present and files in MP4 format, callback in file.getBlob() isn't invoked.

If I remove the code block, everything works as expected. Therefore, for some reason, getBlob() and getBlobURL() don't work well together with appendTo().

**Comment by joshuajazleung** [comment]
I found a solution that works.

You will need to getBlob() and getBlobURL() inside torrent.on('done'). 

Before that, I simply looped through all the files and ran getBlob() and getBlobURL() once the torrent ID was added, hoping the callback would be called when files were finished downloading. Turns out, you will need to run the methods inside torrent.on('done').

I don't know if it's a bug or not, but as long as it's working now :)
